exporters/stdout/stdoutlog/exporter_test.go

```diff
@@ -456,3 +462,493 @@ func TestValueMarshalJSON(t *testing.T) {
+							Data: metricdata.Histogram[float64]{
+								Temporality: metricdata.CumulativeTemporality,
+								DataPoints: []metricdata.HistogramDataPoint[float64]{
+									{
+										Count:      3,
+										Attributes: expectedAttrs,
+									},
+								},
+							},
+						},
+					},
+				}
+
+				metricdatatest.AssertEqual(
+					t,
+					expected,
+					got,
+					metricdatatest.IgnoreTimestamp(),
+					metricdatatest.IgnoreValue(),
+				)
+			},
+		},
+		{
+			name:   "empty_records",
+			enable: true,
+			test: func(t *testing.T, scopeMetrics func() metricdata.ScopeMetrics) {
+				exporter, err := New()
+				require.NoError(t, err)
+				err = exporter.Export(context.Background(), []sdklog.Record{})
+				require.NoError(t, err)
+
+				got := scopeMetrics()
+				assert.Equal(t, "go.opentelemetry.io/otel/exporters/stdout/stdoutlog", got.Scope.Name)
+				assert.NotEmpty(t, got.Scope.Version)
+				assert.NotEmpty(t, got.Metrics, "metrics should be recorded even for empty records")
+
+				assert.Len(t, got.Metrics, 3, "should have 3 metrics for observability")
+				metricNames := make(map[string]bool)
+				for _, metric := range got.Metrics {
+					metricNames[metric.Name] = true
+				}
+				assert.True(t, metricNames["otel.sdk.exporter.log.exported"], "exported metric should be present")
+				assert.True(t, metricNames["otel.sdk.exporter.operation.duration"], "duration metric should be present")
+				assert.True(t, metricNames["otel.sdk.exporter.log.inflight"], "inflight metric should be present")
+			},
+		},
+		{
+			name:   "export_with_error",
+			enable: true,
+			test: func(t *testing.T, scopeMetrics func() metricdata.ScopeMetrics) {
+				exporter, err := New()
+				require.NoError(t, err)
+
+				ctx, cancel := context.WithCancel(context.Background())
+				cancel()
+
+				rf := logtest.RecordFactory{
+					Timestamp: time.Now(),
+					Body:      log.StringValue("test log"),
+				}
+				record := rf.NewRecord()
+				records := []sdklog.Record{record, record}
+
+				err = exporter.Export(ctx, records)
+				require.Error(t, err)
+				require.Equal(t, context.Canceled, err)
+
+				got := scopeMetrics()
+				assert.NotEmpty(t, got.Metrics)
+
+				var exportedMetric metricdata.Metrics
+				exportedInstrument := otelconv.SDKExporterLogExported{}
+				for _, m := range got.Metrics {
+					if m.Name == exportedInstrument.Name() {
+						exportedMetric = m
+						break
+					}
+				}
+				require.NotEmpty(t, exportedMetric, "exported metric not found")
+
+				data := exportedMetric.Data.(metricdata.Sum[int64])
+				require.Len(t, data.DataPoints, 1)
+
+				dataPoint := data.DataPoints[0]
+				assert.Equal(t, int64(2), dataPoint.Value)
+
+				attrs := dataPoint.Attributes
+				errorTypeAttr, found := attrs.Value(attribute.Key("error.type"))
+				assert.True(t, found, "error.type attribute should be present")
+				assert.Equal(t, "*errors.errorString", errorTypeAttr.AsString())
```

Comment:
Validation of the base attributes needs to be done here as well. This will pass if only error attributes are added in an error scenario.